### PR TITLE
Removed root directory methods, added javadoc

### PR DESCRIPTION
**Pull request by gapra-msft**

### CODE REVIEW

File: sdk/storage/azure-storage-file-datalake/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ## 12.0.0-beta.13 (Unreleased)
```
Comment (xiafu-msft) [inline sdk/storage/azure-storage-file-datalake/CHANGELOG.md]: Don't forget to change **## 12.0.0-beta.13 (Unreleased)** before you release ^_^

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileSystemClient.java [hunk 2]
```diff
@@ -129,7 +124,7 @@ public DataLakeDirectoryClient getDirectoryClient(String directoryName) {
      *
      * @return A new {@link DataLakeDirectoryClient} object which references the root directory in this file system.
      */
-    public DataLakeDirectoryClient getRootDirectoryClient() {
```
Comment (xiafu-msft) [inline sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileSystemClient.java]: Do you have tests about getRootDirectoryClient, do you need to edit them?

File: sdk/storage/azure-storage-file-datalake/CHANGELOG.md (hunk 1 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/CHANGELOG.md]: I have another PR for that :P - thanks for checking!

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileSystemClient.java (hunk 2 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileSystemClient.java]: Our tests are in Groovy and it lets you test private methods too - so it shouldnt break them!

**Pull request merged**